packages/react-core/src/components/JumpLinks/JumpLinks.tsx

```diff
@@ -104,47 +104,50 @@ export const JumpLinks: React.FunctionComponent<JumpLinksProps> = ({
   React.useEffect(() => setIsExpanded(isExpandedProp), [isExpandedProp]);
   const navRef = React.useRef<HTMLElement>();
 
-  if (hasScrollSpy) {
-    React.useEffect(() => {
-      if (!canUseDOM) {
-        return;
+  const scrollSpy = React.useCallback(() => {
+    if (!canUseDOM || !hasScrollSpy) {
+      return;
+    }
+    const scrollableElement = document.querySelector(scrollableSelector) as HTMLElement;
```

Comment:
Can we pull this variable out of the `scrollSpy` function and just set it in the `useEffect` below so `scrollSpy` can just use the reference to the queried element?  It shouldn't need to be queried on every scroll action.

Reply:
Good idea, done!